pupa/importers/people.py

```diff
@@ -46,24 +56,58 @@ def limit_spec(self, spec):
 
     def get_object(self, person):
         all_names = [person['name']] + [o['name'] for o in person['other_names']]
-        matches = list(self.model_class.objects.filter(
-            Q(memberships__organization__jurisdiction_id=self.jurisdiction_id),
-            (Q(name__in=all_names) | Q(other_names__name__in=all_names))
-        ).distinct('id'))
-        if len(matches) == 1:
-            return matches[0]
-        elif len(matches) == 0:
-            raise self.model_class.DoesNotExist('No Person: {} in {}'.format(all_names,
-                                                                             self.jurisdiction_id))
+        if person['source_identified']:
+            # special lookup for when we're assuming that, for every combination
+            # of:
+            #
+            #    person.name
+            #    person.source.url
+            #    (optional) person.source.note
+            #
+            # ..there is a unique person in the world that is identified. This
+            # is particularly useful for people who are not well identified by
+            # the data, or, indeed, by any public information.
+
+            spec = {'name__in': all_names, }
+
+            main_query = Q(**spec)
+
+            source_qs = []
+            if len(person['sources']) == 0:
+                raise KeyError('source-identified person {} has no sources!'.format(
```

Comment:
`KeyError` also strikes me as funny - what do we raise if there are no sources anyway? Don't we always require sources?